src/httprpc.cpp

```diff
@@ -228,6 +246,27 @@ static bool InitRPCAuthentication()
     if (gArgs.GetArg("-rpcauth","") != "")
     {
         LogPrintf("Using rpcauth authentication.\n");
+        // Load permissions
+        if(gArgs.GetBoolArg("-rpcwhitelistenable", false)) {
+            std::vector<std::string> permissions = gArgs.GetArgs("-rpcwhitelist");
+            for (unsigned int i = 0; i < permissions.size(); ++i) {
+                // Parse element, skip malformatted elements
+                if(permissions[i].find(':') == std::string::npos) {
```

Comment:
this is incorrect. If a username is given with no trailing semicolon; it should default to all-off.